### ISSUE TITLE
Enquiry about hyper-parameter tuning using sweep

### ISSUE THREAD

**Issue by NancyQuris** [facebookresearch/DomainBed#111]
Hi authors, 

Thank you for providing the community with this fabulous test bed. I found this repo extremely helpful to my research about domain generalization. When reading the code in `sweep.py` and `train.py`, there is a line that I am not very sure about its rationale.

When we tune hyper parameters by calling `sweep.py`, we should keep the generated parameter the same across different trials. However, current seed of random hyper parameter generation in `train.py` is set to be `misc.seed_hash(args.hparams_seed, args.trial_seed)` (line79), which means that the set of hyper parameters will only be given one chance in the run to show its performance during sweep. I am wondering if a seed `misc.seed_hash(args.hparams_seed)` will be enough when we call `train.py` via `sweep.py`, as we should test the randomly generated set of hyper parameters in the same number of trials as we do for the default hyper parameters.

I will really appreciate it if you could enlighten me on my above question about the hyper-parameter tuning in `sweep.py`.

Thank you!

**Comment by qiaoruiyt** [comment]
Thanks Nancy. Same question here! It feels that the same seed (hparams_seed excluding the trial_seed) should be used to set the hyperparameters across the corresponding trials. Otherwise, the resultant mean and error bar would not be for the same set of hyperparameters.

**Comment by lopezpaz** [comment]
@igul222 Can you shed some light on this one?

**Comment by lopezpaz** [comment]
For more explanation, see https://github.com/facebookresearch/DomainBed/issues/103#issuecomment-1330008560
